actions/pacstrap_action.go

```diff
@@ -81,41 +81,54 @@ func (d *PacstrapAction) PreMachine(context *debos.DebosContext, m *fakemachine.
 	return nil
 }
 
-func (d *PacstrapAction) Run(context *debos.DebosContext) error {
-	d.LogStart()
-
-	if d.Mirror == "" {
-		return fmt.Errorf("No mirror set, aborting.")
-	}
-	if len(d.Repositories) == 0 {
-		return fmt.Errorf("No repositories configured.")
-	}
-
-	// Create config for pacstrap
-	configPath := path.Join(context.Scratchdir, "pacman.conf")
+func (d *PacstrapAction) writePacmanConfig(context *debos.DebosContext, configPath string) error {
 	f, err := os.OpenFile(configPath, os.O_RDWR|os.O_CREATE, 0644)
 	if err != nil {
 		return fmt.Errorf("Couldn't open pacman config: %v", err)
 	}
+	defer func() {
+		f.Close()
+		if err != nil {
+			os.Remove(configPath)
+		}
+	}()
+
```

Comment:
Nit: let's make this cleanup (nicely spotted btw) a separate patch. It's a bugfix, while the rest is cosmetics/style.